File: src/main/scala/edu/cnm/deepdive/scala/bitset/Sieve.scala
package edu.cnm.deepdive.scala.bitset

import scala.collection.mutable

object Sieve {

  def sieve(limit: Int): mutable.BitSet = {
    // TODO Implement with a BitSet for keeping track of the candidates.
    throw new UnsupportedOperationException()
  }

  def main(args: Array[String]): Unit = {
    val start = System.currentTimeMillis()
    val upperBound = 1_000_000
    val primes = sieve(upperBound)
    val end = System.currentTimeMillis()
    println(
      s"""Scala Sieve with BitSet and Seq<Int>:
         |${primes.size} primes found between ${primes.firstKey} and ${primes.lastKey} (inclusive) in ${end - start} ms.""".stripMargin
    )
  }

}

File: src/main/java/edu/cnm/deepdive/sieve/array/Sieve.java
package edu.cnm.deepdive.sieve.array;

public class Sieve {

  private static int[] sieve(int limit) {
    // TODO Implement with a boolean[] for keeping track of the candidates, and return an int[] of
    //  all values found to be prime.
    throw new UnsupportedOperationException();
  }

  public static void main(String... args) {
    int upperBound = 1_000_000;
    long start = System.currentTimeMillis();
    int[] primes = sieve(upperBound);
    long end = System.currentTimeMillis();
    System.out.printf("Java Sieve with boolean[] and int[]:%n"
            + "%1$,d primes found between %2$,d and %3$,d in %4$d ms.%n",
        primes.length, primes[0], primes[primes.length - 1], end - start);
  }

}
